### PR TITLE
Chore/build warnings

### PR DESCRIPTION
**Pull request by solarisstar**

### CODE REVIEW

File: code/cgame/cg_drawtools.c [hunk 1]
```diff
@@ -1665,7 +1665,7 @@ qboolean CG_LoadCrosshairs(void) {
     file_len = trap_FS_FOpenFile(fileName, &f, FS_READ);
 
     memset(&charText, 0, sizeof(charText));
-    memset(&cgs.crosshairsData, 0, sizeof(&cgs.crosshairsData));
```
Comment (MJLHThomassen) [inline code/cgame/cg_drawtools.c:1668]: Lol, beginners error XD

File: code/cgame/fx_transporter.c [hunk 2]
```diff
@@ -150,7 +150,7 @@ void FX_SPTransporterLensFlares(centity_t* cent, vec3_t headVector, int startTim
         }
         //Phase 3: flares get smaller
         if ((cg.time < timeOffset + 2000) && (cg.time >= timeOffset + 1500)) {
-            ratio = 1.0f - (float)(cg.time - (timeOffset + 1500)) * 0, 002;
```
Comment (MJLHThomassen) [inline code/cgame/fx_transporter.c:153]: Why did this even build in the first place?

File: code/game/bg_misc.c [hunk 3]
```diff
@@ -1095,17 +1095,11 @@ qboolean	BG_PlayerTouchesItem(playerState_t *ps, entityState_t *item, int atTime
...
-        || ps->origin[2] - origin[2] > 36
-        || ps->origin[2] - origin[2] < -36) {
-        return qfalse;
-    }
+    int xDist = abs(ps->origin[0] - origin[0]);
+    int yDist = abs(ps->origin[1] - origin[1]);
+    int zDist = abs(ps->origin[2] - origin[2]);
 
-    return qtrue;
+    return (xDist < 50 && yDist < 36 && zDist < 36);
```
Comment (MJLHThomassen) [inline code/game/bg_misc.c:1102]: How does this cover ps->origin[0] - origin[0] > 44 ?

**Comment by MJLHThomassen** [comment]
It's been a while using actual pointers and deferences for me, so i had to read up XD.

Code looks good, there is just one thing i commented on in the code which might not be the same after the refactor. I don't know exactly what that code does though.

File: code/game/bg_misc.c (hunk 3 above)
Comment (solarisstar) [inline code/game/bg_misc.c:1102]: I'm thinking that was an oversight on their part. Why should the distance be different forward and back on the x axis?

File: code/cgame/fx_transporter.c (hunk 2 above)
Comment (solarisstar) [inline code/cgame/fx_transporter.c:153]: Better question : why were all the clang warnings ignored?

**Pull request merged**